packages/core/src/resources/Commits.ts

```diff
@@ -195,6 +197,27 @@ export type EditPipelineStatusOptions = {
 };
 
 export class Commits<C extends boolean = false> extends BaseResource<C> {
+  all<E extends boolean = false, P extends PaginationTypes = 'offset'>(
+    projectId: string | number,
+    options: AllCommitsOptions &
+      PaginationRequestOptions<P> &
+      Sudo &
+      ShowExpanded<E> & { withStats: true },
+  ): Promise<GitlabAPIResponse<(CommitSchema & { stats: CommitStatsSchema })[], C, E, P>>;
+
+  all<E extends boolean = false, P extends PaginationTypes = 'offset'>(
+    projectId: string | number,
+    options: AllCommitsOptions &
+      PaginationRequestOptions<P> &
+      Sudo &
+      ShowExpanded<E> & { trailers: true },
+  ): Promise<GitlabAPIResponse<(CommitSchema & { trailers: Record<string, unknown> })[], C, E, P>>;
+
+  all<E extends boolean = false, P extends PaginationTypes = 'offset'>(
+    projectId: string | number,
+    options?: AllCommitsOptions & PaginationRequestOptions<P> & Sudo & ShowExpanded<E>,
+  ): Promise<GitlabAPIResponse<CommitSchema[], C, E, P>>;
+
```

Comment:
There should be another branch to cover the `{ trailers: true, withStats: true }` case.

![image](https://github.com/jdalrymple/gitbeaker/assets/48335/a5d8c06a-bc49-4036-82ba-c0c3dfd5cfef)